src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/SubscriptionRouter.cs

```diff
@@ -15,19 +14,11 @@ public SubscriptionRouter(Publishers publishers, EndpointInstances endpointInsta
             this.transportAddressTranslation = transportAddressTranslation;
         }
 
-        public async Task<IEnumerable<string>> GetAddressesForEventType(Type messageType)
+        public IEnumerable<string> GetAddressesForEventType(Type messageType)
         {
-            var results = new HashSet<string>();
-            foreach (var publisherAddress in publishers.GetPublisherFor(messageType))
-            {
-                var addresses = await publisherAddress.Resolve(endpoint => endpointInstances.FindInstances(endpoint), i => transportAddressTranslation(i)).ConfigureAwait(false);
-                foreach (var address in addresses)
-                {
-                    results.Add(address);
-                }
-            }
-
-            return results;
+            var publisher = publishers.GetPublisherFor(messageType);
```

Comment:
yeah seems like we're missing an acceptance test on that behavior. Will try to add that missing test. 

btw. I think the main reason for that behavior is behavior compatibility with broker transports which essentially support multiple publishers (although we discourage that from a best practice perspective).